ghga_connector/core/batch_processing.py

```diff
@@ -37,9 +40,11 @@ class FileStager:
     staged_files: list[str] = field(default_factory=list, init=False)
     unstaged_files: list[str] = field(default_factory=list, init=False)
 
-    def check_and_stage(self, file_ids: list[str]):
+    def check_and_stage(self, output_dir: Path):
```

Comment:
It might not be a good idea to have the FileStager be responsible for inspecting the output directory. I've putten this to the refactoring list.

Reply:
Factored check out into own class that is injected to handle this.